isomerpages/isomercms-backend
# ref(app): shift support flows into separate folder + ecs service

**NOTE: production deploy stuff is in #1285**

## Problem
At present, some internal flows trigger a [huge spike](https://ogp.datadoghq.com/apm/trace/4475574368042772015?colorBy=service&env=production&graphType=flamegraph&shouldShowLegend=true&sort=time&spanID=718787862051750205&spanViewType=infrastructure&timeHint=1711529331280.001&view=spans) in CPU usage, which will erroneously trigger an alarm. additionally, this CPU spike might cause a degradation on performance of the actual app. 

Closes ISOM-811

## Solution
1. use a separate ecs service to do it. we did **not** go with a sidecar container as was initially suggested. this is because ecs scaling is [done via metrics](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-autoscaling-targettracking.html) and the most fine-grained metric we have at present (that is relevant) is ecs cpu usage _by service_. as the sidecar is still part of the service, this would cause the support container to [trigger scaling](https://docs.aws.amazon.com/AmazonECS/latest/bestpracticesguide/application.html#task-definition) (and still alarms) on the service as a whole, which doesn't solve the problem. 
2. shift shared services out into `/common` -> this is to facilitate sharing of services between `support` and `src`. the new service listens on port 8082 - this is because on local, we want to expose both services ports to the host machine (hence cannot use same port). 
3.  added deployment step for **staging only** for `support` - added 2 more params to the `aws_deploy` workflow to support this. (as service + dockerfile is now different)
4. add `task-def` for `support` - we cannot reuse existing one because it's a new ecs service. 
 
**Breaking Changes**



  - not user facing, but the existing forms will **not** work and will require an update to `<stg | prod>-support-api/v2/infra/formsg`

## Tests
These tests have been done on `staging`, but the success has been checked **only via email** (ie, receive success email = good enough) 
1. submit the staging site launch form
2. should receive an email regarding the dns records to update
3. submit the staging site creation form
4. should receive a success email
5. submit staging audit logs form
6. should receive success email
7. submit staging site link checker form
8. should have logs (`"*link*"` in `isomer-infra` **or** `isomer` (not entirely sure why, suspect it's because our project tag in infra repo is `isomer-infra` which leads to the tag being `isomer-infra`)
![Screenshot 2024-04-04 at 1 40 03 PM](https://github.com/isomerpages/isomercms-backend/assets/44049504/7808288e-f2b2-4bba-ac49-5467b5efcb07)
9. submit staging site repair form 
10. should see success email
